// playwright-xray/lib/convertToMultipart.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.verifyMultipatConfig = exports.convertToMultipart = void 0;
async function convertToMultipart(testResult, multiPartOptions) {
    const summary = testResult.info.summary;
    const testPlanKey = testResult.info.testPlanKey;
    delete testResult.info.project;
    const info = {
        fields: {
            project: { id: multiPartOptions.project?.id },
            summary: summary,
            issuetype: { id: multiPartOptions.issuetype?.id },
            components: multiPartOptions.components === undefined ? [] : multiPartOptions.components,
        },
        xrayFields: {
            testPlanKey: testPlanKey,
            environments: multiPartOptions.xrayFields?.environments === undefined ? [] : multiPartOptions.xrayFields.environments,
        },
    };
    return { info: JSON.stringify(info), testResult: JSON.stringify(testResult) };
}
exports.convertToMultipart = convertToMultipart;
function verifyMultipatConfig(options) {
    if (options.multiPart === undefined) {
        throw new Error("Multipart options must be provided when useMultipart is true");
    }
    if (options.multiPart.project === undefined || options.multiPart.issuetype === undefined) {
        throw new Error("Multipart options must include project and issuetype");
    }
    if (options.multiPart.xrayFields === undefined) {
        throw new Error("Multipart options must include xrayFields");
    }
    if (options.multiPart.multiPartUrl === undefined) {
        throw new Error("Multipart options must include multiPartUrl");
    }
}
exports.verifyMultipatConfig = verifyMultipatConfig;
